### PR TITLE
Toggleable bool column

### PR DESCRIPTION
**Pull request by felbalart**
Adds addon that features switches to toggle boolean attributes values directly at the index

### CODE REVIEW

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      success: function(data) {
+        boolSwitch.addClass('hidden-switch');
+        otherSwitch.removeClass('hidden-switch');
+        if(!boolSwitch.hasClass('notify-success')) return;
+          $(function() {
+             var successMsg = 'Update Successful!';
+            setTimeout(alert(successMsg), 500)
+          });
+      },
+      type: 'PATCH'
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Missing trailing comma                          comma-dangle

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        var errorMsg = "Error: Update Unsuccessful";
+        alert(errorMsg);
+      },
+      success: function(data) {
+        boolSwitch.addClass('hidden-switch');
+        otherSwitch.removeClass('hidden-switch');
+        if(!boolSwitch.hasClass('notify-success')) return;
+          $(function() {
+             var successMsg = 'Update Successful!';
+            setTimeout(alert(successMsg), 500)
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Unexpected alert                                no-alert<br>Missing semicolon                               semi

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      error: function() {
+        var errorMsg = "Error: Update Unsuccessful";
+        alert(errorMsg);
+      },
+      success: function(data) {
+        boolSwitch.addClass('hidden-switch');
+        otherSwitch.removeClass('hidden-switch');
+        if(!boolSwitch.hasClass('notify-success')) return;
+          $(function() {
+             var successMsg = 'Update Successful!';
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Expected indentation of 12 spaces but found 13  indent

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      dataType: 'json',
+      error: function() {
+        var errorMsg = "Error: Update Unsuccessful";
+        alert(errorMsg);
+      },
+      success: function(data) {
+        boolSwitch.addClass('hidden-switch');
+        otherSwitch.removeClass('hidden-switch');
+        if(!boolSwitch.hasClass('notify-success')) return;
+          $(function() {
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Expected indentation of 8 spaces but found 10   indent

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      data: data,
+      dataType: 'json',
+      error: function() {
+        var errorMsg = "Error: Update Unsuccessful";
+        alert(errorMsg);
+      },
+      success: function(data) {
+        boolSwitch.addClass('hidden-switch');
+        otherSwitch.removeClass('hidden-switch');
+        if(!boolSwitch.hasClass('notify-success')) return;
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Expected space(s) after "if"                    keyword-spacing

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    data[model][field] = otherValue;
+    $.ajax({
+      url: url,
+      data: data,
+      dataType: 'json',
+      error: function() {
+        var errorMsg = "Error: Update Unsuccessful";
+        alert(errorMsg);
+      },
+      success: function(data) {
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: 'data' is defined but never used                no-unused-vars

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    data = { id: objectId };
+    data[model] = {};
+    data[model][field] = otherValue;
+    $.ajax({
+      url: url,
+      data: data,
+      dataType: 'json',
+      error: function() {
+        var errorMsg = "Error: Update Unsuccessful";
+        alert(errorMsg);
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Unexpected alert                                no-alert

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    var url = boolSwitch.data('url');
+    data = { id: objectId };
+    data[model] = {};
+    data[model][field] = otherValue;
+    $.ajax({
+      url: url,
+      data: data,
+      dataType: 'json',
+      error: function() {
+        var errorMsg = "Error: Update Unsuccessful";
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Strings must use singlequote                    quotes

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    var otherValue = String(value)!=='true';
+    var switchId = boolSwitch.attr('id');
+    var otherSwitch = $('#' + switchId.substr(0, switchId.lastIndexOf("-") + 1) + otherValue);
+    var url = boolSwitch.data('url');
+    data = { id: objectId };
+    data[model] = {};
+    data[model][field] = otherValue;
+    $.ajax({
+      url: url,
+      data: data,
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: 'data' is not defined                           no-undef

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    var objectId = boolSwitch.data('object_id');
+    var field = boolSwitch.data('field');
+    var value = boolSwitch.data('value');
+    var otherValue = String(value)!=='true';
+    var switchId = boolSwitch.attr('id');
+    var otherSwitch = $('#' + switchId.substr(0, switchId.lastIndexOf("-") + 1) + otherValue);
+    var url = boolSwitch.data('url');
+    data = { id: objectId };
+    data[model] = {};
+    data[model][field] = otherValue;
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: 'data' is not defined                           no-undef

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+        var errorMsg = 'Error: Update Unsuccessful';
+        alert(errorMsg);
+      },
+      success: function() {
+        boolSwitch.addClass('hidden-switch');
+        otherSwitch.removeClass('hidden-switch');
+        if (!boolSwitch.hasClass('notify-success')) return;
+        $(function() {
+          var successMsg = 'Update Successful!';
+          setTimeout(alert(successMsg), 500);
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Unexpected alert                              no-alert

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+    var data = { id: objectId };
+    data[model] = {};
+    data[model][field] = otherValue;
+    $.ajax({
+      url: url,
+      data: data,
+      dataType: 'json',
+      error: function() {
+        var errorMsg = 'Error: Update Unsuccessful';
+        alert(errorMsg);
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Unexpected alert                              no-alert

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 13]
```diff
@@ -0,0 +1,35 @@
+$(function() {
+  $('.toggle-bool-switch').click(function (e) {
+    var boolSwitch = $(e.target);
+    var model = boolSwitch.data('model');
+    var objectId = boolSwitch.data('object_id');
+    var field = boolSwitch.data('field');
+    var value = boolSwitch.data('value');
+    var otherValue = String(value)!=='true';
+    var switchId = boolSwitch.attr('id');
+    var otherSwitch = $('#' + switchId.substr(0, switchId.lastIndexOf("-") + 1) + otherValue);
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Strings must use singlequote                  quotes

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 14]
```diff
@@ -0,0 +1,35 @@
+$(function() {
+  $('.toggle-bool-switch').click(function (e) {
+    var boolSwitch = $(e.target);
+    var model = boolSwitch.data('model');
+    var objectId = boolSwitch.data('object_id');
+    var field = boolSwitch.data('field');
+    var value = boolSwitch.data('value');
+    var otherValue = String(value)!=='true';
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Infix operators must be spaced                space-infix-ops

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 15]
```diff
@@ -0,0 +1,35 @@
+$(function() {
+  $('.toggle-bool-switch').click(function (e) {
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Unexpected space before function parentheses  space-before-function-paren

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+        var errorMsg = 'Error: Update Unsuccessful';
+        alert(errorMsg);
+      },
+      success: function() {
+        boolSwitch.addClass('hidden-switch');
+        otherSwitch.removeClass('hidden-switch');
+        if (!boolSwitch.hasClass('notify-success')) return;
+        $(function() {
+          var successMsg = 'Update Successful!';
+          setTimeout(alert(successMsg), 500);
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Unexpected alert                no-alert

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+    var data = { id: objectId };
+    data[model] = {};
+    data[model][field] = otherValue;
+    $.ajax({
+      url: url,
+      data: data,
+      dataType: 'json',
+      error: function() {
+        var errorMsg = 'Error: Update Unsuccessful';
+        alert(errorMsg);
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Unexpected alert                no-alert

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 18]
```diff
@@ -0,0 +1,35 @@
+$(function() {
+  $('.toggle-bool-switch').click(function(e) {
+    var boolSwitch = $(e.target);
+    var model = boolSwitch.data('model');
+    var objectId = boolSwitch.data('object_id');
+    var field = boolSwitch.data('field');
+    var value = boolSwitch.data('value');
+    var otherValue = String(value)!== 'true';
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Infix operators must be spaced  space-infix-ops

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 19]
```diff
@@ -0,0 +1,35 @@
...
+        var errorMsg = 'Error: Update Unsuccessful';
+        alert(errorMsg);
+      },
+      success: function() {
+        boolSwitch.addClass('hidden-switch');
+        otherSwitch.removeClass('hidden-switch');
+        if (!boolSwitch.hasClass('notify-success')) return;
+        $(function() {
+          var successMsg = 'Update Successful!';
+          setTimeout(alert(successMsg), 500);
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Unexpected alert  no-alert

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 20]
```diff
@@ -0,0 +1,35 @@
...
+    var data = { id: objectId };
+    data[model] = {};
+    data[model][field] = otherValue;
+    $.ajax({
+      url: url,
+      data: data,
+      dataType: 'json',
+      error: function() {
+        var errorMsg = 'Error: Update Unsuccessful';
+        alert(errorMsg);
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Unexpected alert  no-alert

File: docs/toggle_bool.md [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+ 
+ `toggle_bool_column :paid, unless: proc {|item| item.is_free? }`
+ 
+ ### Notify result
+ 
+ Optionally a js alert dialog can be prompted to the user upon update success.
+ 
+ This is disabled by default, but can be enabled with the option `notify`
+   
+  `toggle_bool_column :paid, notify: true`
```
Comment (ldlsegovia) [inline docs/toggle_bool.md]: @felbalart  Entiendo que el mensaje de error esté hardcodeado porque en realidad no debería pasar nunca, pero el de success creo que debería poder manejarse con locales. En este sentido, me parece que sería más útil un `success_msg: I18n.t('bla')` en lugar de un `notify: true`. Se comportaría igual... Si no está definido `success_msg`, no se muestra.

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 22]
```diff
@@ -0,0 +1,36 @@
...
+      },
+      success: function() {
+        boolSwitch.addClass('hidden-switch');
+        otherSwitch.removeClass('hidden-switch');
+        if (!boolSwitch.hasClass('notify-success')) return;
+        $(function() {
+          var successMsg = 'Update Successful!';
+          setTimeout(alert(successMessage), 500);
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Unexpected alert                                 no-alert

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+      error: function() {
+        var errorMsg = 'Error: Update Unsuccessful';
+        alert(errorMsg);
+      },
+      success: function() {
+        boolSwitch.addClass('hidden-switch');
+        otherSwitch.removeClass('hidden-switch');
+        if (!boolSwitch.hasClass('notify-success')) return;
+        $(function() {
+          var successMsg = 'Update Successful!';
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: 'successMsg' is assigned a value but never used  no-unused-vars

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+    var data = { id: objectId };
+    data[model] = {};
+    data[model][field] = otherValue;
+    $.ajax({
+      url: url,
+      data: data,
+      dataType: 'json',
+      error: function() {
+        var errorMsg = 'Error: Update Unsuccessful';
+        alert(errorMsg);
```
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Unexpected alert                                 no-alert

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js (hunk 22 above)
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Unexpected alert                                 no-alert

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js (hunk 23 above)
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: 'successMsg' is assigned a value but never used  no-unused-vars

File: app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js (hunk 24 above)
Comment (monkeyci) [inline app/assets/javascripts/activeadmin_addons/addons/toggle_bool.js]: Unexpected alert                                 no-alert

**Pull request merged**